src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2667,7 +2666,7 @@ void os::jvm_path(char *buf, jint buflen) {
 
         // determine if this is a legacy image or modules image
         // modules image doesn't have "jre" subdirectory
-        len = strlen(buf);
+        len = (int)strlen(buf);
```

Comment:
Why raw cast here when other uses of strlen have had checked_cast applied?

Reply:
Most of the vm code casts strlen result to int because it's always an int.  strlen returning size_t for a small buffer is really annoying.  Maybe a checked_cast<> would catch overflow though and maybe it's a good idea.  It's weird how fgets takes int and not size_t.